src/core/lib/transport/byte_stream.cc

```diff
@@ -55,8 +55,11 @@ void SliceBufferByteStream::Orphan() {
 
 bool SliceBufferByteStream::Next(size_t /*max_size_hint*/,
                                  grpc_closure* /*on_complete*/) {
-  GPR_DEBUG_ASSERT(backing_buffer_.count > 0);
-  return true;
+  if (cursor_ < backing_buffer_.count) {
```

Comment:
This change doesn't look right to me.  A `SliceBufferByteStream` should always return true from `Next()`, because the bytes are always available.

As I mentioned elsewhere, `ByteStream::Next()` should never be called when there is no data left on the stream to read (which is what happens when you read the last slice out of the slice buffer), so there should be no case where this returns false.

I think that if you change the cronet code as I suggested, then the change in this file will not be necessary.

Reply:
Missed this file in my review. Yea I agree with Mark that I don't think it is right to change the behavior of the `Next()` function.

Reply:
done